lib/withOnyx.js

```diff
@@ -132,15 +155,14 @@ export default function (mapOnyxToState) {
                     return;
                 }
 
-                // Leave untouched previous state to avoid data loss during pre-load updates.
-                // This handles case when setState was called before the setWithOnyxState.
-                // For example, when an Onyx property was updated by keyChanged before the call of the setWithOnyxState.
-                this.setState((prevState) => {
-                    const remainingTempState = _.omit(this.tempState, _.keys(prevState));
```

Comment:
I don't quite understand how is this change correct, what if a new value is set for a key in the prevState? Then it will be ommitted?

Reply:
I _think_ the idea is that there can be a race between a new value calling `setState()` on `withOnyx()` and the time it takes to get this initial value from storage. 

And if that happens then we should use the most recent value (the one that was set and not the stale value that we read). 

Tbh - not in love with how this was implemented. I would prefer find some way to mark the data as stale if something changed it's value, but I guess we decided the correct place to do that would be in the `setState()` callback.

Reply:
Ah, now I got it. Well, this is no longer correct if `initialValue` is passed, it will never update because the component starts with a value on that key.

I need to think about how to solve this.